sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/options/GcsOptions.java

```diff
@@ -208,4 +222,13 @@ public PathValidator create(PipelineOptions options) {
           .build();
     }
   }
+
+  @FunctionalInterface
+  public interface GoogleCloudStorageProvider {
+    GoogleCloudStorage get(
+        GoogleCloudStorageOptions options,
+        Storage storage,
+        Credentials credentials,
+        HttpRequestInitializer httpRequestInitializer);
```

Comment:
These 4 params should cover both the [2.x constructor](https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/v2.2.26/gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java#L302-L394) and the [3.x Builder](https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/v3.0.4/gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java#L2496-L2517)